qmail-smtpd.c

```diff
@@ -217,14 +221,44 @@ int flagbarf; /* defined if seenmail */
 stralloc mailfrom = {0};
 stralloc rcptto = {0};
 
+#ifdef SMTPUTF8
+stralloc mfparms = {0};
+void mailfrom_parms(const char *arg)
+{
+  int i;
+  int len;
+
+  len = str_len(arg);
+  mfparms.len=0;
+  i = byte_chr(arg,len,'>');
+  if (i > 4 && i < len) {
+    while (len) {
+      arg++; len--;
+      if (*arg == ' ' || *arg == '\0' ) {
+         if (case_starts(mfparms.s,"SMTPUTF8")) smtputf8 = 1;
+         mfparms.len=0;
+      } else
+        if (!stralloc_catb(&mfparms,arg,1)) die_nomem();
+    }
+  }
+}
+#endif
+
```

Comment:
So far, it appears the only use of this function (which I recognize from Arnt's patch) is to check whether the peer is trying to request `SMTPUTF8` for this delivery. Let's name it accordingly and mark it `static`. If we see other uses for it later, we can generalize then.

Reply:
The same function can be used for getting the ESMTP AUTH and SIZE parameters. I will change this to static for the moment.